### PR TITLE
[3672] Drag and Drop issues

### PR DESCRIPTION
**Pull request by aachavarria**
Adding notify to pointer-controller
fixing issue with browse for shared components.
https://github.com/craftercms/craftercms/issues/3672

### CODE REVIEW

File: static-assets/components/cstudio-forms/controls/node-selector.js [hunk 1]
```diff
@@ -511,8 +511,12 @@ YAHOO.extend(CStudioForms.Controls.NodeSelector, CStudioForms.CStudioFormField,
             }
 
             item.datasource = datasource;
-            this.items[this.items.length] = item;
-
+            if (order) {
```
Comment (rart) [inline static-assets/components/cstudio-forms/controls/node-selector.js]: Order = 0 will go to else too. Test specifically for what you need `!= null` (single equals) for example will catch what you seem to want here.

File: static-assets/components/cstudio-preview-tools/mods/component-panel.js [hunk 2]
```diff
@@ -202,10 +202,12 @@
                     if (response.supported) {
                       if (isNew) {
                         function isNewEvent(value, modelPath) {
+                          debugger;
```
Comment (rart) [inline static-assets/components/cstudio-preview-tools/mods/component-panel.js]: x

File: static-assets/components/cstudio-preview-tools/mods/component-panel.js [hunk 3]
```diff
@@ -225,7 +227,8 @@
                           //If no path provided the dnd item is a embedded content
                           if (!path) {
                             var selectorId = Object.keys(zones)[0];
-                            ComponentsPanel.onDropEmbedded(previewPath, compPath, type, selectorId, response.ds);
+                            var order = zones[selectorId].findIndex((item) => item == tracking);
```
Comment (rart) [inline static-assets/components/cstudio-preview-tools/mods/component-panel.js]: ===

File: static-assets/scripts/pointer-controller.js [hunk 4]
```diff
@@ -112,11 +112,9 @@ crafterDefine('pointer-controller', ['crafter', 'jquery', 'jquery-ui', 'animator
       $(divMouse).css('top', e.pageY);
     });
     try {
-      $(window.parent.document).keyup(function (e) {
-        var _self = this;
+      $window.keyup(function (e) {
```
Comment (rart) [inline static-assets/scripts/pointer-controller.js]: Need to remove listener once it executes or that the operation is finished.

File: static-assets/scripts/pointer-controller.js [hunk 5]
```diff
@@ -139,14 +137,15 @@ crafterDefine('pointer-controller', ['crafter', 'jquery', 'jquery-ui', 'animator
...
-        componentDropped.call(me, $dropZone, $component);
+        componentDropped.call(me, $dropZone, $component, destinationZone);
       } else {
         me.done();
         publish.call(me, Topics.START_DIALOG, {
           message: 'The component cannot be added, it is already in the drop-zone.',
-          height: '208px'
+          height: '248px'
         });
         //return;
```
Comment (rart) [inline static-assets/scripts/pointer-controller.js]: x

**Review by rart — CHANGES_REQUESTED** [review]

**Review by rart — APPROVED** [review]

**Review by russdanner — COMMENTED** [review]
Looks self contained and code looks good.

**Pull request merged**